core/src/main/java/org/apache/spark/memory/MemoryConsumer.java

```diff
@@ -41,7 +41,7 @@ protected MemoryConsumer(TaskMemoryManager taskMemoryManager, long pageSize, Mem
   }
 
   protected MemoryConsumer(TaskMemoryManager taskMemoryManager) {
```

Comment:
When will we construct a `MemoryConsumer` while the memory mode is different from `TaskMemoryManager`'s memory mode?